### ISSUE TITLE
origin-docker-builder cgroup limit on Debian based systems

### ISSUE THREAD

**Issue by mattirk** [openshift/origin#7297]
Cgroup CPU limits seem to be broken as of v1.1.2 version of origin-docker-builder. The builder tries to get CPU limits from:

/sys/fs/cgroup/cpuacct,cpu/cpu.cfs_quota_us
but the file resides at:
/sys/fs/cgroup/cpu/cpu.cfs_quota_us
in Debian based systems.

Error:
1 builder.go:202]  Error: failed to retrieve cgroup limits: open /sys/fs/cgroup/cpuacct,cpu/cpu.cfs_quota_us: no such file or directory

Utility function causing the error:
https://github.com/openshift/origin/blob/master/pkg/build/builder/util.go#L70

**Comment by derekwaynecarr** [comment]
Can you confirm that this is fixed now by https://github.com/openshift/origin/pull/7181 ?

**Comment by mattirk** [comment]
I did another test with commit 9e9cb3d and v1.1.2 using Ubuntu 14.04.3 LTS and the same error occurs:

`builder.go:202] Error: failed to retrieve cgroup limits: open /sys/fs/cgroup/cpuacct,cpu/cpu.cfs_quota_us: no such file or directory`

I did another test by swapping binaries to ones from build v1.1.1 and the Docker build runs without problems.

**Comment by bparees** [comment]
v1.1.1 didn't attempt to read those values.
v1.1.2 did attempt to read them, but didn't include the fix to check additional paths like the one debian uses.

i'm not sure what you did when you tried to test w/ commit 9e9cb3d, but the changes are in the builder images themselves (openshift/origin-sti-builder and openshift/origin-docker-builder) so you need to make sure you were using the updated versions of those images, not just an updated openshift binary.

you can do a docker pull on those images (to all your nodes) and then run openshift start with "--latest-images" to allow it to use those newly pulled images.

**Comment by mattirk** [comment]
Both STI and Docker builders indeed word with latest images. Hopefully a new tagged version with this issue fixed will be released soon. Using latest tagged images is always a bit scary.

**Comment by bparees** [comment]
@smarterclayton are we going to do a v1.1.2.1 anytime soon?  there are a number of significant build issues (docker builds can't pull secure images using secrets, can't build on debian due to cgroups paths) in v1.1.2.

(i'd also like to get https://github.com/openshift/origin/pull/7184 in before we do one, though)

**Comment by smarterclayton** [comment]
I'll just do a 1.1.3.  Tell me when 7184 lands

**Comment by harobed** [comment]
Same error here :

« builder.go:204] Error: failed to retrieve cgroup limits: open /sys/fs/cgroup/cpu/cpu.cfs_quota_us: no such file or directory »

```
# docker version
Client:
 Version:      1.10.1
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   9e83765
 Built:        Thu Feb 11 19:09:42 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.1
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   9e83765
 Built:        Thu Feb 11 19:09:42 2016
 OS/Arch:      linux/amd64
```

and 

```
# openshift version
openshift v1.1.3
kubernetes v1.2.0-origin
etcd 2.2.2+git
```

```
# cat /etc/debian_version
8.3
```

```
# ls /sys/fs/cgroup/ -1
blkio
cpu
cpuacct
cpu,cpuacct
cpuset
devices
freezer
memory
net_cls
net_cls,net_prio
net_prio
perf_event
systemd
root@debian-jessie-build:/sys/fs/cgroup/cpu# ls /sys/fs/cgroup/cpu/ -1
cgroup.clone_children
cgroup.procs
cgroup.sane_behavior
cpuacct.stat
cpuacct.usage
cpuacct.usage_percpu
cpu.shares
docker
docker-daemon
kubelet
notify_on_release
release_agent
tasks
```

**Comment by harobed** [comment]
```
# uname  --all
Linux debian-jessie-build 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt20-1+deb8u3 (2016-01-17) x86_64 GNU/Linux
```

**Comment by bparees** [comment]
@harobed this was fixed in 1.1.4:
https://github.com/openshift/origin/blob/master/pkg/build/builder/util.go#L93
